File: Python/String-2/cat_dog.py
# Return True if the string "cat" and "dog" appear 
# the same number of times in the given string.


# cat_dog('catdog') → True
# cat_dog('catcat') → False
# cat_dog('1cat1cadodog') → True
def cat_dog(str):
  cc = 0
  dc = 0
  for i in range(len(str)-2):
    if str[i:i+3] == "cat":
      cc = cc+1
    elif str[i:i+3] == "dog":
      dc=dc+1
  return True if cc == dc else False

File: java/String-2/catDog.java
// Return true if the string "cat" and "dog" appear the same number of times in the given string.

// catDog("catdog") → true
// catDog("catcat") → false
// catDog("1cat1cadodog") → true

public boolean catDog(String str) {
  int cc = 0, dc = 0;
  for(int i = 0; i < str.length()-2; i++) {
    if(str.substring(i,i+3).equals("cat")) cc++;
    else if(str.substring(i,i+3).equals("dog")) dc++;
  }
  return cc == dc;
}
